File: backend/app/schemas/f1.py
from datetime import date, datetime

from pydantic import BaseModel


class DriverOut(BaseModel):
    id: int
    driver_ref: str
    code: str | None
    number: int | None
    full_name: str
    nationality: str | None
    headshot_url: str | None

    model_config = {"from_attributes": True}


class ConstructorOut(BaseModel):
    id: int
    constructor_ref: str
    name: str
    nationality: str | None
    color: str | None
    logo_url: str | None = None

    model_config = {"from_attributes": True}


class CircuitOut(BaseModel):
    id: int
    circuit_ref: str
    name: str
    location: str | None
    country: str | None
    lat: float | None
    lng: float | None
    length_km: float | None
    corners: int | None
    drs_zones: int | None
    track_type: str | None
    first_gp_year: int | None
    lap_record_time: str | None
    lap_record_driver: str | None
    lap_record_year: int | None

    model_config = {"from_attributes": True}


class DriverStandingOut(BaseModel):
    position: int
    points: float
    wins: int
    podiums: int
    driver: DriverOut
    constructor: ConstructorOut | None
    last_five: list[str]  # finish positions, newest last ("1", "DNF", ...)


class ConstructorStandingOut(BaseModel):
    position: int
    points: float
    wins: int
    podiums: int
    constructor: ConstructorOut
    drivers: list[DriverOut]
    last_five_points: list[float]  # points per round, newest last


class SessionSlot(BaseModel):
    name: str
    starts_at: datetime | None


class NextRaceOut(BaseModel):
    race_id: int
    season: int
    round: int
    name: str
    date: date
    time: str | None
    circuit: CircuitOut
    schedule: list[SessionSlot]
    previous_winner: str | None
    previous_pole: str | None
    previous_fastest_lap: str | None


class SeasonProgressOut(BaseModel):
    season: int
    total_rounds: int
    completed_rounds: int
    leader: DriverStandingOut | None
    leading_constructor: ConstructorStandingOut | None
    next_race: NextRaceOut | None


class ProgressPoint(BaseModel):
    round: int
    race_name: str
    points: float  # cumulative


class ProgressSeries(BaseModel):
    entity_id: int
    label: str
    color: str | None
    points: list[ProgressPoint]


class ChampionshipProgressOut(BaseModel):
    season: int
    entity_type: str  # "driver" | "constructor"
    series: list[ProgressSeries]


class TrendingStat(BaseModel):
    key: str
    label: str
    holder: str
    detail: str | None
    value: str
    color: str | None


class PodiumEntry(BaseModel):
    position: int
    driver: DriverOut
    constructor: ConstructorOut | None
    time_text: str | None


class RaceSummaryOut(BaseModel):
    race_id: int
    season: int
    round: int
    name: str
    date: date
    circuit: CircuitOut
    podium: list[PodiumEntry]
    fastest_lap_driver: str | None
    fastest_lap_time: str | None
    biggest_gainer: str | None
    biggest_gain: int | None
    pole_sitter: str | None


# --- Driver analytics (Phase 2) ---


class DriverListItem(BaseModel):
    driver: DriverOut
    constructor: ConstructorOut | None  # most recent within the scope
    season: int | None  # null for all-time career scope
    races: int
    wins: int
    podiums: int
    poles: int
    points: float
    best_finish: int | None


class DriverRating(BaseModel):
    key: str
    label: str
    value: float  # 0–100 normalised score
    detail: str  # human-readable underlying statistic


class DriverSeasonStat(BaseModel):
    season: int
    constructor: ConstructorOut | None
    races: int
    points: float
    wins: int
    podiums: int
    poles: int
    best_finish: int | None
    championship_position: int | None


class DriverCareerStats(BaseModel):
    seasons: int
    first_season: int | None
    last_season: int | None
    races: int
    wins: int
    podiums: int
    poles: int
    fastest_laps: int
    points: float
    dnfs: int
    titles: int
    best_championship_position: int | None
    win_rate: float  # 0–1
    podium_rate: float  # 0–1
    avg_grid: float | None
    avg_finish: float | None


class DriverResultLine(BaseModel):
    season: int
    round: int
    race_name: str
    circuit: str
    grid: int | None
    position: int | None
    position_text: str | None
    points: float
    status: str | None


class DriverDetailOut(BaseModel):
    driver: DriverOut
    dob: date | None
    current_constructor: ConstructorOut | None
    teams: list[ConstructorOut]
    career: DriverCareerStats
    ratings: list[DriverRating]
    seasons: list[DriverSeasonStat]
    strengths: list[str]
    weaknesses: list[str]
    recent_results: list[DriverResultLine]
    data_since: int | None  # earliest ingested season (stats coverage caveat)


# --- Constructor analytics (Phase 2) ---


class ConstructorDriver(BaseModel):
    driver: DriverOut
    races: int
    seasons: int
    is_current: bool


class ConstructorListItem(BaseModel):
    constructor: ConstructorOut
    season: int | None
    races: int
    wins: int
    podiums: int
    poles: int
    points: float
    championship_position: int | None
    drivers: list[DriverOut]  # lineup within the scope (carries headshots)


class ConstructorSeasonStat(BaseModel):
    season: int
    constructor: ConstructorOut | None  # the team itself (drives chart colour)
    races: int
    points: float
    wins: int
    podiums: int
    poles: int
    best_finish: int | None
    championship_position: int | None


class ConstructorCareerStats(BaseModel):
    seasons: int
    first_season: int | None
    last_season: int | None
    races: int
    entries: int  # individual car entries
    wins: int
    podiums: int
    poles: int
    one_twos: int
    fastest_laps: int
    points: float
    dnfs: int
    titles: int
    best_championship_position: int | None
    win_rate: float
    podium_rate: float
    avg_grid: float | None
    avg_finish: float | None


class ConstructorResultLine(BaseModel):
    season: int
    round: int
    race_name: str
    best_position: int | None
    points: float
    car_results: list[str]  # e.g. ["VER P1", "LAW R"]


class ConstructorDetailOut(BaseModel):
    constructor: ConstructorOut
    career: ConstructorCareerStats
    ratings: list[DriverRating]
    seasons: list[ConstructorSeasonStat]
    drivers: list[ConstructorDriver]  # lineup + alumni, most recent first
    strengths: list[str]
    weaknesses: list[str]
    recent_results: list[ConstructorResultLine]
    data_since: int | None


# --- Circuit analytics (Phase 2) ---


class CircuitListItem(BaseModel):
    circuit: CircuitOut
    races_held: int  # completed GPs in the ingested data
    first_year: int | None
    last_year: int | None
    on_current_calendar: bool
    last_winner: str | None


class CircuitWinnerLine(BaseModel):
    season: int
    race_name: str
    driver: DriverOut
    constructor: ConstructorOut | None
    grid: int | None
    time_text: str | None


class TopEntry(BaseModel):
    """A 'most successful here' row (driver or constructor) with win count."""

    label: str
    color: str | None
    value: int


class CircuitDetailOut(BaseModel):
    circuit: CircuitOut
    races_held: int
    first_year: int | None
    last_year: int | None
    on_current_calendar: bool
    next_race_date: date | None
    distinct_winners: int
    pole_win_rate: float | None  # share of races won from pole (0–1)
    dnf_rate: float | None  # share of car entries not classified (0–1)
    avg_finishers: float | None  # classified cars per race
    top_drivers: list[TopEntry]
    top_constructors: list[TopEntry]
    winners: list[CircuitWinnerLine]  # most recent first
    data_since: int | None


# --- Calendar / history explorer (Phase 2) ---


class CalendarRace(BaseModel):
    race_id: int
    season: int
    round: int
    name: str
    date: date
    time: str | None
    circuit: CircuitOut
    completed: bool
    winner: str | None
    winner_color: str | None
    pole_sitter: str | None


class RaceResultRow(BaseModel):
    position: int | None
    position_text: str | None
    driver: DriverOut
    constructor: ConstructorOut | None
    grid: int | None
    points: float
    status: str | None
    fastest_lap: bool
    time_text: str | None


class RaceDetailOut(BaseModel):
    race_id: int
    season: int
    round: int
    name: str
    date: date
    circuit: CircuitOut
    pole_sitter: str | None
    fastest_lap_driver: str | None
    results: list[RaceResultRow]


# --- Driver vs driver comparison (Phase 2) ---


class HeadToHead(BaseModel):
    shared_races: int
    a_race_ahead: int  # times A finished ahead of B (both classified)
    b_race_ahead: int
    a_quali_ahead: int
    b_quali_ahead: int


class CompareOut(BaseModel):
    a: DriverDetailOut
    b: DriverDetailOut
    head_to_head: HeadToHead


# --- Race predictor + championship simulation (Phase 3) ---


class PredictionFactor(BaseModel):
    text: str  # human-readable stat behind the prediction
    positive: bool  # whether it pushes the win probability up or down


class RacePredictionEntry(BaseModel):
    driver: DriverOut
    constructor: ConstructorOut | None
    grid: int | None
    win_prob: float  # normalised share of the field's win probability (sums to 1)
    podium_prob: float  # raw model probability
    top10_prob: float
    factors: list[PredictionFactor]
    actual_position: int | None  # populated for completed races (backtest view)
    actual_text: str | None  # "R"/"D"/"W" when the real result was a non-finish


class PredictModelInfo(BaseModel):
    algorithm: str
    trained_at: str | None
    train_seasons: str | None
    holdout_seasons: str | None
    holdout_races: int | None
    auc_win: float | None
    auc_podium: float | None
    winner_pick_rate: float | None  # holdout share of races where top pick won
    features: list[str]


class RacePredictionOut(BaseModel):
    race_id: int
    season: int
    round: int
    name: str
    date: date
    circuit: CircuitOut
    completed: bool
    grid_source: str  # "official" | "qualifying" | "estimated"
    certainty: float  # 0-1, entropy-based spread of the win probabilities
    model: PredictModelInfo
    entries: list[RacePredictionEntry]


class TitleContender(BaseModel):
    driver: DriverOut
    constructor: ConstructorOut | None
    current_points: float
    current_position: int | None
    title_prob: float
    top3_prob: float
    expected_points: float


class ChampionshipSimOut(BaseModel):
    season: int
    completed_rounds: int
    total_rounds: int
    remaining_sprints: int
    iterations: int
    contenders: list[TitleContender]


# --- Global search (Phase 2) ---


class SearchDriver(BaseModel):
    ref: str
    name: str
    code: str | None
    nationality: str | None
    headshot_url: str | None


class SearchConstructor(BaseModel):
    ref: str
    name: str
    color: str | None
    logo_url: str | None


class SearchCircuit(BaseModel):
    ref: str
    name: str
    location: str | None
    country: str | None


class SearchOut(BaseModel):
    drivers: list[SearchDriver]
    constructors: list[SearchConstructor]
    circuits: list[SearchCircuit]

File: frontend/src/types/f1.ts
/** Mirrors backend/app/schemas/f1.py */

export interface Driver {
  id: number;
  driver_ref: string;
  code: string | null;
  number: number | null;
  full_name: string;
  nationality: string | null;
  headshot_url: string | null;
}

export interface Constructor {
  id: number;
  constructor_ref: string;
  name: string;
  nationality: string | null;
  color: string | null;
  logo_url: string | null;
}

export interface Circuit {
  id: number;
  circuit_ref: string;
  name: string;
  location: string | null;
  country: string | null;
  lat: number | null;
  lng: number | null;
  length_km: number | null;
  corners: number | null;
  drs_zones: number | null;
  track_type: string | null;
  first_gp_year: number | null;
  lap_record_time: string | null;
  lap_record_driver: string | null;
  lap_record_year: number | null;
}

export interface DriverStanding {
  position: number;
  points: number;
  wins: number;
  podiums: number;
  driver: Driver;
  constructor: Constructor | null;
  last_five: string[];
}

export interface ConstructorStanding {
  position: number;
  points: number;
  wins: number;
  podiums: number;
  constructor: Constructor;
  drivers: Driver[];
  last_five_points: number[];
}

export interface SessionSlot {
  name: string;
  starts_at: string | null;
}

export interface NextRace {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  time: string | null;
  circuit: Circuit;
  schedule: SessionSlot[];
  previous_winner: string | null;
  previous_pole: string | null;
  previous_fastest_lap: string | null;
}

export interface SeasonProgress {
  season: number;
  total_rounds: number;
  completed_rounds: number;
  leader: DriverStanding | null;
  leading_constructor: ConstructorStanding | null;
  next_race: NextRace | null;
}

export interface ProgressPoint {
  round: number;
  race_name: string;
  points: number;
}

export interface ProgressSeries {
  entity_id: number;
  label: string;
  color: string | null;
  points: ProgressPoint[];
}

export interface ChampionshipProgress {
  season: number;
  entity_type: "driver" | "constructor";
  series: ProgressSeries[];
}

export interface TrendingStat {
  key: string;
  label: string;
  holder: string;
  detail: string | null;
  value: string;
  color: string | null;
}

export interface PodiumEntry {
  position: number;
  driver: Driver;
  constructor: Constructor | null;
  time_text: string | null;
}

export interface RaceSummary {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  circuit: Circuit;
  podium: PodiumEntry[];
  fastest_lap_driver: string | null;
  fastest_lap_time: string | null;
  biggest_gainer: string | null;
  biggest_gain: number | null;
  pole_sitter: string | null;
}

export interface HealthStatus {
  status: string;
  database: string;
  version: string;
}

// --- Driver analytics (Phase 2) ---

export interface DriverListItem {
  driver: Driver;
  constructor: Constructor | null;
  season: number | null;
  races: number;
  wins: number;
  podiums: number;
  poles: number;
  points: number;
  best_finish: number | null;
}

export interface DriverRating {
  key: string;
  label: string;
  value: number;
  detail: string;
}

export interface DriverSeasonStat {
  season: number;
  constructor: Constructor | null;
  races: number;
  points: number;
  wins: number;
  podiums: number;
  poles: number;
  best_finish: number | null;
  championship_position: number | null;
}

export interface DriverCareerStats {
  seasons: number;
  first_season: number | null;
  last_season: number | null;
  races: number;
  wins: number;
  podiums: number;
  poles: number;
  fastest_laps: number;
  points: number;
  dnfs: number;
  titles: number;
  best_championship_position: number | null;
  win_rate: number;
  podium_rate: number;
  avg_grid: number | null;
  avg_finish: number | null;
}

export interface DriverResultLine {
  season: number;
  round: number;
  race_name: string;
  circuit: string;
  grid: number | null;
  position: number | null;
  position_text: string | null;
  points: number;
  status: string | null;
}

export interface DriverDetail {
  driver: Driver;
  dob: string | null;
  current_constructor: Constructor | null;
  teams: Constructor[];
  career: DriverCareerStats;
  ratings: DriverRating[];
  seasons: DriverSeasonStat[];
  strengths: string[];
  weaknesses: string[];
  recent_results: DriverResultLine[];
  data_since: number | null;
}

// --- Constructor analytics (Phase 2) ---

export interface ConstructorListItem {
  constructor: Constructor;
  season: number | null;
  races: number;
  wins: number;
  podiums: number;
  poles: number;
  points: number;
  championship_position: number | null;
  drivers: Driver[];
}

// Structurally identical to DriverSeasonStat, so it reuses SeasonPointsChart.
export interface ConstructorSeasonStat {
  season: number;
  constructor: Constructor | null;
  races: number;
  points: number;
  wins: number;
  podiums: number;
  poles: number;
  best_finish: number | null;
  championship_position: number | null;
}

export interface ConstructorCareerStats {
  seasons: number;
  first_season: number | null;
  last_season: number | null;
  races: number;
  entries: number;
  wins: number;
  podiums: number;
  poles: number;
  one_twos: number;
  fastest_laps: number;
  points: number;
  dnfs: number;
  titles: number;
  best_championship_position: number | null;
  win_rate: number;
  podium_rate: number;
  avg_grid: number | null;
  avg_finish: number | null;
}

export interface ConstructorDriver {
  driver: Driver;
  races: number;
  seasons: number;
  is_current: boolean;
}

export interface ConstructorResultLine {
  season: number;
  round: number;
  race_name: string;
  best_position: number | null;
  points: number;
  car_results: string[];
}

export interface ConstructorDetail {
  constructor: Constructor;
  career: ConstructorCareerStats;
  ratings: DriverRating[];
  seasons: ConstructorSeasonStat[];
  drivers: ConstructorDriver[];
  strengths: string[];
  weaknesses: string[];
  recent_results: ConstructorResultLine[];
  data_since: number | null;
}

// --- Circuit analytics (Phase 2) ---

export interface CircuitListItem {
  circuit: Circuit;
  races_held: number;
  first_year: number | null;
  last_year: number | null;
  on_current_calendar: boolean;
  last_winner: string | null;
}

export interface CircuitWinnerLine {
  season: number;
  race_name: string;
  driver: Driver;
  constructor: Constructor | null;
  grid: number | null;
  time_text: string | null;
}

export interface TopEntry {
  label: string;
  color: string | null;
  value: number;
}

export interface CircuitDetail {
  circuit: Circuit;
  races_held: number;
  first_year: number | null;
  last_year: number | null;
  on_current_calendar: boolean;
  next_race_date: string | null;
  distinct_winners: number;
  pole_win_rate: number | null;
  dnf_rate: number | null;
  avg_finishers: number | null;
  top_drivers: TopEntry[];
  top_constructors: TopEntry[];
  winners: CircuitWinnerLine[];
  data_since: number | null;
}

// --- Calendar / history (Phase 2) ---

export interface CalendarRace {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  time: string | null;
  circuit: Circuit;
  completed: boolean;
  winner: string | null;
  winner_color: string | null;
  pole_sitter: string | null;
}

export interface RaceResultRow {
  position: number | null;
  position_text: string | null;
  driver: Driver;
  constructor: Constructor | null;
  grid: number | null;
  points: number;
  status: string | null;
  fastest_lap: boolean;
  time_text: string | null;
}

export interface RaceDetail {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  circuit: Circuit;
  pole_sitter: string | null;
  fastest_lap_driver: string | null;
  results: RaceResultRow[];
}

// --- Compare (Phase 2) ---

export interface HeadToHead {
  shared_races: number;
  a_race_ahead: number;
  b_race_ahead: number;
  a_quali_ahead: number;
  b_quali_ahead: number;
}

export interface CompareResult {
  a: DriverDetail;
  b: DriverDetail;
  head_to_head: HeadToHead;
}

// --- Race predictor + championship simulation (Phase 3) ---

export interface PredictionFactor {
  text: string;
  positive: boolean;
}

export interface RacePredictionEntry {
  driver: Driver;
  constructor: Constructor | null;
  grid: number | null;
  win_prob: number;
  podium_prob: number;
  top10_prob: number;
  factors: PredictionFactor[];
  actual_position: number | null;
  actual_text: string | null;
}

export interface PredictModelInfo {
  algorithm: string;
  trained_at: string | null;
  train_seasons: string | null;
  holdout_seasons: string | null;
  holdout_races: number | null;
  auc_win: number | null;
  auc_podium: number | null;
  winner_pick_rate: number | null;
  features: string[];
}

export interface RacePrediction {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  circuit: Circuit;
  completed: boolean;
  grid_source: "official" | "qualifying" | "estimated";
  certainty: number;
  model: PredictModelInfo;
  entries: RacePredictionEntry[];
}

export interface TitleContender {
  driver: Driver;
  constructor: Constructor | null;
  current_points: number;
  current_position: number | null;
  title_prob: number;
  top3_prob: number;
  expected_points: number;
}

export interface ChampionshipSim {
  season: number;
  completed_rounds: number;
  total_rounds: number;
  remaining_sprints: number;
  iterations: number;
  contenders: TitleContender[];
}

// --- Global search (Phase 2) ---

export interface SearchDriver {
  ref: string;
  name: string;
  code: string | null;
  nationality: string | null;
  headshot_url: string | null;
}

export interface SearchConstructor {
  ref: string;
  name: string;
  color: string | null;
  logo_url: string | null;
}

export interface SearchCircuit {
  ref: string;
  name: string;
  location: string | null;
  country: string | null;
}

export interface SearchResults {
  drivers: SearchDriver[];
  constructors: SearchConstructor[];
  circuits: SearchCircuit[];
}

// --- Strategy simulator (Phase 3) ---
export interface StrategyCircuitListItem {
  circuit_ref: string;
  name: string;
  country: string | null;
  laps: number;
  base_lap_s: number;
  pit_loss_s: number;
  on_calendar: boolean;
  has_pace: boolean;
}

export interface StrategyCompound {
  key: string;
  name: string;
  color: string;
  offset: number;
  deg: number;
}

export interface StrategyStint {
  compound: string;
  name: string;
  color: string;
  start_lap: number;
  end_lap: number;
  laps: number;
}

export interface StrategyOption {
  key: string;
  label: string;
  stops: number;
  stints: StrategyStint[];
  pits: number[];
  compound_sequence: string[];
  total_time_s: number;
  total_time_str: string;
  avg_lap_s: number;
  lap_pace: number[];
  delta_s: number;
}

export interface StrategyCircuitInfo {
  circuit_ref: string;
  name: string;
  country: string | null;
  laps: number;
  base_lap_s: number;
  base_lap_str: string;
  pit_loss_s: number;
  deg_mode: string;
  real_avg_stops: number | null;
  calibrated: boolean;
}

export interface StrategySim {
  circuit: StrategyCircuitInfo;
  compounds: StrategyCompound[];
  optimal_key: string;
  strategies: StrategyOption[];
}

// --- Track suitability (Phase 3) ---
export interface SuitabilityEntry {
  driver: Driver;
  constructor: Constructor | null;
  score: number;
  here_starts: number;
  here_best: number | null;
  reason: string;
}

export interface SuitabilityResult {
  circuit_ref: string;
  track_type: string | null;
  entries: SuitabilityEntry[];
}

// --- Weather + tyre analytics (Phase 3, FastF1) ---
export interface RaceConditionLine {
  season: number;
  race_name: string;
  air_temp: number | null;
  track_temp: number | null;
  rainfall: boolean;
  wet_fraction: number;
}

export interface CircuitConditions {
  circuit_ref: string;
  races_with_data: number;
  avg_air_temp: number | null;
  avg_track_temp: number | null;
  wet_races: number;
  wet_rate: number | null;
  recent: RaceConditionLine[];
}

export interface WetRatingEntry {
  driver: Driver;
  constructor: Constructor | null;
  wet_races: number;
  wet_avg_finish: number;
  dry_avg_finish: number | null;
  delta: number;
}

export interface WetRatings {
  wet_race_count: number;
  entries: WetRatingEntry[];
}

export interface CompoundUsage {
  compound: string;
  color: string;
  stints: number;
  laps: number;
  share: number;
  avg_stint_laps: number;
  avg_pace_s: number | null;
  deg_s_per_lap: number | null;
}

export interface CircuitTyres {
  circuit_ref: string;
  races_with_data: number;
  compounds: CompoundUsage[];
}

export interface TyreOverview {
  races_with_data: number;
  compounds: CompoundUsage[];
}

// --- Auth + profiles (Phase 4) ---
export type FavoriteKind = "driver" | "constructor" | "circuit";

export interface Favorite {
  entity_type: FavoriteKind;
  entity_ref: string;
}

export interface AuthUser {
  id: number;
  email: string;
  username: string;
  display_name: string | null;
  theme: string;
  created_at: string;
  favorites: Favorite[];
}

export interface AuthToken {
  access_token: string;
  token_type: string;
  user: AuthUser;
}

// --- Prediction League (Phase 4) ---
export type LeagueRaceStatus = "open" | "awaiting" | "completed";

export interface LeagueDriverOption {
  driver: Driver;
  constructor: Constructor | null;
}

export interface PredictionPicks {
  pole_driver_id: number | null;
  winner_driver_id: number | null;
  p2_driver_id: number | null;
  p3_driver_id: number | null;
  fastest_lap_driver_id: number | null;
}

export type SubmitPrediction = PredictionPicks & { race_id: number };

export interface ScoreBreakdown {
  pole: number;
  winner: number;
  podium: number;
  fastest_lap: number;
  total: number;
}

export interface LeagueRaceItem {
  race_id: number;
  season: number;
  round: number;
  name: string;
  date: string;
  time: string | null;
  circuit: Circuit;
  status: LeagueRaceStatus;
  predicted: boolean;
  your_points: number | null;
}

export interface LeagueRaces {
  season: number;
  max_points: number;
  races: LeagueRaceItem[];
}

export interface LeagueRaceDetail {
  race: LeagueRaceItem;
  options: LeagueDriverOption[];
  prediction: PredictionPicks | null;
  actual: PredictionPicks | null;
  score: ScoreBreakdown | null;
}

export interface LeaderboardEntry {
  rank: number;
  user_id: number;
  username: string;
  display_name: string | null;
  total_points: number;
  scored: number;
  is_you: boolean;
}

export interface Leaderboard {
  entries: LeaderboardEntry[];
  your_rank: number | null;
}

export interface MyPredictionItem {
  race: LeagueRaceItem;
  picks: PredictionPicks;
  score: ScoreBreakdown | null;
}

export interface MyPredictions {
  total_points: number;
  scored: number;
  items: MyPredictionItem[];
}

// --- AI Assistant (Phase 4) ---
export interface AssistantStat {
  label: string;
  value: string;
}

export interface AssistantEntity {
  kind: FavoriteKind;
  ref: string;
  name: string;
}

export interface AssistantAnswer {
  answer: string;
  intent: string;
  provider: string;
  stats: AssistantStat[];
  entities: AssistantEntity[];
  suggestions: string[];
}
